### ISSUE TITLE
Chart: no point annotations or text callouts on data

### ISSUE THREAD

**Issue by ryanoneill** [ryanoneill/envision#368]
## Summary

There is no way to annotate specific data points or regions on a chart with text labels. You can add vertical reference lines and horizontal thresholds, but you cannot:

- Label a specific point ("Best epoch: 42, loss=0.031")
- Add a text callout to a region ("Overfitting begins")
- Mark a point with a distinct symbol (star, diamond, arrow)
- Add a text box with an arrow pointing to a feature of interest

## Why this matters

Annotations are what turn a chart from "data display" into "data communication." Researchers annotate charts constantly:

- Marking the best validation loss and its epoch
- Labeling phase transitions (memorization → grokking → generalization)
- Pointing out anomalies ("GPU thermal throttling here")
- Marking intervention points ("learning rate reduced", "data augmentation added")
- Comparing key values ("89.2% vs 87.1%")

Vertical lines (with labels) partially address this, but they only mark x-positions — they can't point to a specific (x, y) coordinate, can't show values, and don't support free-text annotations positioned near data features.

## Suggested approach

```rust
let chart = ChartState::line(series)
    .with_annotation(42.0, 0.031, "Best: 0.031", Color::Green)
    .with_annotation(75.0, 1.2, "Overfitting begins", Color::Yellow);
```

Rendering: Place the text near the specified (x, y) coordinate, offset slightly to avoid overlapping the data point. Use a simple positioning heuristic (above-right by default, shift if near edges). Optionally render a small marker (●, ★) at the exact coordinate.

The text should be styled distinctly from axis labels (perhaps bold, or with a background highlight) so it reads as an annotation rather than chart chrome.
